packages/react/src/components/TreeView/TreeNode.tsx

```diff
@@ -328,43 +369,104 @@ const TreeNode = React.forwardRef<HTMLLIElement, TreeNodeProps>(
     };
 
     if (!children) {
+      if (href) {
+        return (
+          <li role="none">
+            <Link
+              {...treeNodeProps}
```

Comment:
Something that @mbgower surfaced in slack was that the link should be **wrapped** in the role of treeitem:

> Okay, so you’re just using a link mechansim to populate the main content area, and I assume that link is wrapped in the role of treeitem so that it works in the tree as expected.
> @tay1orjones overall i can’t think of an issue with this approach.

Here `treeNodeProps` contains the `role="treeitem"`, so it's not wrapped but placed directly on the link.

Reply:
I confirmed with @mbgower just now that this isn't a concern.